main.cpp

```diff
@@ -169,7 +176,39 @@ static int parse_argv(int argc, char *argv[], const char **uart)
         return -EINVAL;
     }
 
-    *uart = argv[optind++];
+    if (stat(argv[optind], &st) == -1) {
```

Comment:
It's already a TOCTOU. It's an argument the user is passing, there isn't much option.